app/models/models.py

```diff
@@ -0,0 +1,151 @@
+from datetime import datetime, timedelta
+from decimal import Decimal
+from math import floor
+from typing import Any, Dict, List, Set
+
+from config import (
+	DATE_TIME_FORMAT,
+	FREE_MINUTES,
+	TIME_FRAME_HOURS,
+	TIME_FRAME_TARIFFS
+)
+
+class ParkingLotFullError(Exception):
+	"""Custom error that is raised when the Parking Lot is full"""
+	pass
+
+class ParkingLotLocationEmptyError(Exception):
+	"""Custom error that is raised when the Parking Lot location is empty"""
+	pass
+
+class Ticket:
+	def __init__(self, license_plate: str, tariff: str, location: int) -> None:
+		"""Ticket constructor
+
+		Args:
+			license_plate (str): Car license plate.
+			tariff (str): Tariff for the ticket.
+			location (int): Parking Lot location.
+
+		Returns:
+			None
+		"""
+		self.car: str = license_plate
+		self.tariff: str = tariff
+		self.location: int = location
+		self.start: str = datetime.now().strftime(DATE_TIME_FORMAT)
+		self.finish: str = ""
+		self.fee: str = ""
+
+	def _get_total_time(self) -> timedelta:
+		"""Get the total time for the ticket
+
+		Returns:
+			timedelta: Total time for the ticket.
+		"""
+		try:
+			start = datetime.strptime(self.start, DATE_TIME_FORMAT)
+			finish = datetime.strptime(self.finish, DATE_TIME_FORMAT)
+		except ValueError as e:
+			raise ValueError("Invalid date format") from e
```

Comment:
Do we really need to convert timestamp to string to later parse it?

Reply:
Fixed. Datetimes are only converted to a string when sending back the JSON response because a datetime object is not JSON serializable.